iree/turbine/kernel/wave/schedule_reordering.py

```diff
@@ -301,37 +470,135 @@ def transform_two_PP_clusters(
     # 1st cluster interleaved local and global reads.
     clusters.append(sliced_local_load_lhs[0])
     clusters.append(sliced_local_load_rhs[0])
-    clusters.append(SchedulingBarrier([]).add_to_graph(tmp_graph))
+    clusters.append(
+        insert_op_after(
```

Comment:
Why do we need this insert_op_after? If you place the scheduling barrier, then won't the topological_sort_with_dependencies take care of it?

Reply:
I see, it is for more fine grained control.